Fixed thorium duping via fluid extractor

supermattdab0ss opened the pull request:
The way nuclearcraft isotopes work is a natural ingot (144mb) is split into a clump (144mb) and small clump (16mb), both oredicted as a ingot and nugget. Gregtech sees these oredicts, and decides to add recipes in the fluid extractor for thorium, even though these aren't supposed to be recombined into natural thorium. The script removes these recipes, and adds a new one in the mixer to recombine the correct isotopes into thorium dust.

As per usual in the modded minecraft world, law of conservation of mass is a joke.

- fixed #228

supermattdab0ss committed:
Fixed thorium duping via fluid extractor

The way nuclearcraft isotopes work is a natural ingot (144mb) is split into a clump (144mb) and small clump (16mb), both oredicted as a ingot and nugget. Gregtech sees these oredicts, and decides to add recipe in the fluid extractor for thorium, even though these aren't supposed to be recombined into natural thorium. The script removes these recipes, and adds a new one in the mixer to recombine the correct isotopes into thorium dust.

Exaxxion commented on overrides/scripts/Nuclearcraft.zs:
```diff
@@ -2,9 +2,9 @@ import crafttweaker.item.IItemStack;
-
 import mods.gregtech.recipe.RecipeMap;
 
+val fluid_extractor = RecipeMap.getByName("fluid_extractor");
```
This variable is defined but not used.

There is a globally defined reference to the fluid extractor recipe registry in [CommonVars.zs](https://github.com/Exaxxion/Omnifactory/blob/1.3-PR/overrides/scripts/CommonVars.zs) which is loaded first by CraftTweaker so we don't need to redefine the machines in each file. I had not cleaned up all of the redundant variables in the script files, so sorry if that was confusing.

supermattdab0ss replied:
Fixed in latest commit. Better make sure at some point we don't have GT scripts that load before CommonVars though.

supermattdab0ss committed:
Removing repeated GT machine vars

Removed the duplicate variables as they are already present as global variables.

ALongStringOfNumbers commented:
This PR is superseded by #588
